site/content/docs/developing/addons/workload.en.md

```diff
@@ -0,0 +1,240 @@
+* If you'd like to inject a secret to your ECS task, you can define a [Secret](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-secretsmanager-secret.html) in your template, and then add it as an [Output](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html). The secret will be injected into your container and can be accessed as an environment variable as capital SNAKE_CASE.
+* If you'd like to inject any resource value as an environment variable, you can create an [Output](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html) for any value that you want to be injected as an environment variable to your ECS tasks. It will be injected into your container and accessed as an environment variable  as capital SNAKE_CASE.
+
+## Examples
+
+### A Workload Addon Template For A DynamoDB Table
+
+Here is an example template for a workload-level DynamoDB table addon:
+```yaml
+# You can use any of these parameters to create conditions or mappings in your template.
+Parameters:
+  App:
+    Type: String
+    Description: Your application's name.
+  Env:
+    Type: String
+    Description: The environment name your service, job, or workflow is being deployed to.
+  Name:
+    Type: String
+    Description: The name of the service, job, or workflow being deployed.
+
+Resources:
+  # Create your resource here, such as an AWS::DynamoDB::Table:
+  # MyTable:
+  #   Type: AWS::DynamoDB::Table
+  #   Properties:
+  #     ...
+
+  # 1. In addition to your resource, if you need to access the resource from your ECS task 
+  # then you need to create an AWS::IAM::ManagedPolicy that holds the permissions for your resource.
+  #
+  # For example, below is a sample policy for MyTable:
+  MyTableAccessPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: DDBActions
+            Effect: Allow
+            Action:
+              - dynamodb:BatchGet*
+              - dynamodb:DescribeStream
+              - dynamodb:DescribeTable
+              - dynamodb:Get*
+              - dynamodb:Query
+              - dynamodb:Scan
+              - dynamodb:BatchWrite*
+              - dynamodb:Create*
+              - dynamodb:Delete*
+              - dynamodb:Update*
+              - dynamodb:PutItem
+            Resource: !Sub ${ MyTable.Arn}
+
+Outputs:
+  # 1. You need to output the IAM ManagedPolicy so that Copilot can add it as a managed policy to your ECS task role.
+  MyTableAccessPolicyArn:
+    Description: "The ARN of the ManagedPolicy to attach to the task role."
+    Value: !Ref MyTableAccessPolicy
+
+  # 2. If you want to inject a property of your resource as an environment variable to your ECS task,
+  # then you need to define an output for it.
+  #
+  # For example, the output MyTableName will be injected in capital snake case, MY_TABLE_NAME, to your task.
```

Comment:
elsewhere, we use `SCREAMING_SNAKE_CASE`...should we be consistent?

Reply:
`MY_TABLE_NAME` is the exact env var name that would be injected into the container. iirc `SCREAMING_SNAKE_CASE` is used as a general example. So I'd say we probably want to keep it here.

Reply:
Oh, I just meant "screaming snake case" vs "capital snake case." 🤷🏼 I guess it doesn't really matter if we use both!